### DISCUSSION TITLE
Handling a polar singular with conservative regridding?

### DISCUSSION THREAD

**Discussion by michaelerb** [pangeo-data/xESMF#325] [General]
Hi.  I am trying to regrid some model data, but I'm having trouble at the poles.  The model data is on a rectangular lat/lon grid with latitude values which range from -90 to 90.  All values at 90 are the same (ditto -90), as would be expected, since they're all the same location.  When I try to regrid using conservative regridding, I get a message that my latitude values are outside of [-90,90], and the regridded values at the poles are wrong.  They're approximately twice the expected value.  Perhaps the code thinks that the gridboxes at the poles should actually span the pole, thereby doubling them up?  At any rate, what the best solution to this?  Is there an option I should be using when regridded?  Or should I just not be using the conservative regridding?  Thanks!

**Comment by raphaeldussin** [comment]
@michaelerb there are a few options to deal with the poles. first make sure you use periodic=True then try ignore_degenerate=True

**Comment by michaelerb** [comment]
Hi @raphaeldussin.  Thanks for the suggestions.  Unfortunately, neither one fixed the problem.  Are there any other suggestions?

**Comment by raphaeldussin** [comment]
can you post a small example (ideally that do not need real data) that reproduces the problem?

**Comment by sol1105** [comment]
Hi @michaelerb, this might be the same issue I encountered here #157, which I could avoid by providing the bounds for the input dataset myself (so they would not be calculated implicitly and possibly exceed [-90, 90] degrees latitude) or using the `conservative_normed` remapping method, or using the option `skip_na=True` when applying the weights. However the last two options affect the entire regridded dataset.

**Comment by michaelerb** [comment]
Thanks for the suggestions @sol1105.  The "conservative_normed" option seems to work, although I don't quite understand why.  I haven't tried the other options yet, but I will soon.  @sol1105 or @raphaeldussin: For global climate model output (quantities like monthly temperature or precipitation), is there a recommended approach?

To provide more information about my original question, here's some python code which reproduces the problem I described using sample data.  I've also attached an image of the figure this code produces.  As you can see, the original data has a value of 1 everywhere, but the values at the north and south poles are around 2 (although not exactly).

# Code to reproduce problem
```
import numpy as np
import xarray as xr
import xesmf as xe
import matplotlib.pyplot as plt
import cartopy.crs as ccrs

#%% CREATE DATA
lat_orig = np.arange(-90,91,2)
lon_orig = np.arange(0,360,2)
sample_data = np.ones((len(lat_orig),len(lon_orig)))
data_xarray = xr.Dataset(
    {
         'sample_data':(['lat','lon'],sample_data)
    },
    coords={
        'lat': (['lat'],lat_orig,{'units':'degrees_north'}),
        'lon': (['lon'],lon_orig,{'units':'degrees_east'}),
    },
)

#%% REGRIDDING
#Set up output variable
lat_new = np.arange(-89,90,2)
lon_new = np.linspace(0,360,136)[:-1]
data_format = xr.Dataset(
    {
        'lat': (['lat'],lat_new,{'units':'degrees_north'}),
        'lon': (['lon'],lon_new,{'units':'degrees_east'}),
    }
)

#Set up the regridder and do the regridding
regridder = xe.Regridder(data_xarray,data_format,'conservative')
#regridder = xe.Regridder(data_xarray,data_format,'conservative_normed')
#regridder = xe.Regridder(data_xarray,data_format,'conservative',periodic=True,ignore_degenerate=True)
#regridder = xe.Regridder(data_xarray,data_format,'bilinear',periodic=True)
data_regridded = regridder(data_xarray,keep_attrs=True)

#%% MAKE A FIGURE
plt.figure(figsize=(12,12))
ax1 = plt.subplot2grid((2,1),(0,0),projection=ccrs.PlateCarree())
ax2 = plt.subplot2grid((2,1),(1,0),projection=ccrs.PlateCarree())
data_xarray.sample_data.plot.pcolormesh(ax=ax1)
data_regridded.sample_data.plot.pcolormesh(ax=ax2)
ax1.set_title('Original')
ax2.set_title('Regridded')
ax1.coastlines()
ax2.coastlines()
plt.plot()
```

# Output of code above
![regridding](https://user-images.githubusercontent.com/15681231/158915387-82367aee-6264-4d74-957a-dc7ee021d038.png)
